packages/go/analysis/ad/adcscache.go

```diff
@@ -91,51 +98,132 @@ func (s ADCSCache) BuildCache(ctx context.Context, db graph.Database, enterprise
 			} else if authStoreForNodes, err := FetchEnterpriseCAsTrustedForNTAuthToDomain(tx, domain); err != nil {
 				log.Errorf("Error getting cas via authstorefor for domain %d: %v", domain.ID, err)
 			} else {
-				s.AuthStoreForChainValid[domain.ID] = cardinality.NodeSetToDuplex(authStoreForNodes)
-				s.RootCAForChainValid[domain.ID] = cardinality.NodeSetToDuplex(rootCaForNodes)
+				s.authStoreForChainValid[domain.ID] = cardinality.NodeSetToDuplex(authStoreForNodes)
+				s.rootCAForChainValid[domain.ID] = cardinality.NodeSetToDuplex(rootCaForNodes)
 			}
 
 			// Check for weak cert config on DCs
-			if upnMapping, err := HasUPNCertMappingInForest(tx, domain); err != nil {
-				log.Warnf("Error checking HasUPNCertMappingInForest for domain %d: %v", domain.ID, err)
+			if upnMapping, err := hasUPNCertMappingInForest(tx, domain); err != nil {
+				log.Warnf("Error checking hasUPNCertMappingInForest for domain %d: %v", domain.ID, err)
 				return nil
 			} else if upnMapping {
-				s.HasUPNCertMappingInForest[domain.ID] = struct{}{}
+				s.hasUPNCertMappingInForest.Add(domain.ID.Uint32())
 			}
-			if weakCertBinding, err := HasWeakCertBindingInForest(tx, domain); err != nil {
-				log.Warnf("Error checking HasWeakCertBindingInForest for domain %d: %v", domain.ID, err)
+			if weakCertBinding, err := hasWeakCertBindingInForest(tx, domain); err != nil {
+				log.Warnf("Error checking hasWeakCertBindingInForest for domain %d: %v", domain.ID, err)
 				return nil
 			} else if weakCertBinding {
-				s.HasWeakCertBindingInForest[domain.ID] = struct{}{}
+				s.hasWeakCertBindingInForest.Add(domain.ID.Uint32())
 			}
 		}
 
 		return nil
 	})
+	if err != nil {
+		log.Errorf("Error building adcs cache %v", err)
+	}
 
 	log.Infof("Finished building adcs cache")
 }
 
-func (s ADCSCache) DoesCAChainProperlyToDomain(enterpriseCA, domain *graph.Node) bool {
+func (s *ADCSCache) DoesCAChainProperlyToDomain(enterpriseCA, domain *graph.Node) bool {
 	var domainID = domain.ID
 	var caID = enterpriseCA.ID.Uint32()
 
-	if _, ok := s.RootCAForChainValid[domainID]; !ok {
+	s.RLock()
+	defer s.RUnlock()
+	if _, ok := s.rootCAForChainValid[domainID]; !ok {
 		return false
-	} else if _, ok := s.AuthStoreForChainValid[domainID]; !ok {
+	} else if _, ok := s.authStoreForChainValid[domainID]; !ok {
 		return false
 	} else {
-		return s.RootCAForChainValid[domainID].Contains(caID) && s.AuthStoreForChainValid[domainID].Contains(caID)
+		return s.rootCAForChainValid[domainID].Contains(caID) && s.authStoreForChainValid[domainID].Contains(caID)
 	}
 }
 
-func HasUPNCertMappingInForest(tx graph.Transaction, domain *graph.Node) (bool, error) {
+func (s *ADCSCache) GetExpandedCertTemplateControllers(id graph.ID) (cardinality.Duplex[uint32], bool) {
```

Comment:
General question, not a fix suggestion: if all we do with the boolean return is assign it to the no-name variable `_` then should we just refactor these to return empty cardinality duplexes?

Reply:
Same applies to all incarnations of this pattern in this changeset.